frontend/src/metabase/query_builder/components/view/ViewHeader/components/QuestionActions/QuestionMoreActionsMenu.tsx

```diff
@@ -0,0 +1,220 @@
+      datasetEditorTab: "query",
+    });
+
+  const handleEditMetadata = () =>
+    onSetQueryBuilderMode("dataset", {
+      datasetEditorTab: "metadata",
+    });
+
+  const handleTurnToModel = () => {
+    const modal = checkCanBeModel(question)
+      ? MODAL_TYPES.TURN_INTO_DATASET
+      : MODAL_TYPES.CAN_NOT_CREATE_MODEL;
+    trackTurnIntoModelClicked(question);
+    onOpenModal(modal);
+  };
+  const onOpenSettingsSidebar = () => dispatch(onOpenQuestionSettings());
+
+  const onTurnModelIntoQuestion = () => dispatch(turnModelIntoQuestion());
+
+  const label = isDashboardQuestion
+    ? t`Move, duplicate, and more…`
+    : t`Move, trash, and more…`;
+
+  const menuItems = [
+    (isStandaloneQuestion || isMetric) && (
+      <Menu.Item
+        key="add_to_dash"
+        icon={<Icon name="add_to_dash" />}
+        onClick={() => onOpenModal(MODAL_TYPES.ADD_TO_DASHBOARD)}
+        data-testid={ADD_TO_DASH_TESTID}
+      >
+        {t`Add to dashboard`}
+      </Menu.Item>
+    ),
+    ...PLUGIN_MODERATION.useQuestionMenuItems(question, reload),
+    hasCollectionPermissions && isModelOrMetric && hasDataPermissions && (
+      <Menu.Item
+        key="edit_definition"
+        icon={<Icon name="notebook" />}
+        onClick={handleEditQuery}
+      >
+        {isMetric ? t`Edit metric definition` : t`Edit query definition`}
+      </Menu.Item>
+    ),
+    hasCollectionPermissions && isModel && (
+      <Menu.Item
+        key="edit-metadata"
+        icon={<Icon name="label" />}
+        data-testid="edit-metadata"
+        onClick={handleEditMetadata}
+      >
+        <div>
+          {t`Edit metadata`}{" "}
+          <DatasetMetadataStrengthIndicator
+            className={QuestionActionsS.StrengthIndicator}
+            dataset={question}
+          />
+        </div>
+      </Menu.Item>
+    ),
+    hasCollectionPermissions && !isDashboardQuestion && !isModel && (
+      <Menu.Item
+        key="turn_into_model"
+        icon={<Icon name="model" />}
+        data-testid={TURN_INTO_DATASET_TESTID}
+        onClick={handleTurnToModel}
+      >
+        {t`Turn into a model`}
+      </Menu.Item>
+    ),
+    hasCollectionPermissions && isModel && (
+      <Menu.Item
+        key="turn_into_question"
+        icon={<Icon name="insight" />}
+        onClick={onTurnModelIntoQuestion}
+      >
+        {t`Turn back to saved question`}
+      </Menu.Item>
+    ),
+    enableSettingsSidebar && (
+      <Menu.Item
+        key="edit-settings"
+        icon={<Icon name="gear" />}
+        data-testid="question-settings-button"
+        onClick={onOpenSettingsSidebar}
+      >
+        {t`Edit settings`}
+      </Menu.Item>
+    ),
+    hasCollectionPermissions && (
+      <>
+        <Menu.Divider />
+        <Menu.Item
+          key="move"
+          icon={<Icon name="move" />}
+          data-testid={MOVE_TESTID}
+          onClick={() => onOpenModal(MODAL_TYPES.MOVE)}
+        >
+          {t`Move`}
```

Comment:
As above, a context string would be helpful here. If you want to use one, make sure it's exactly the same here as above to avoid creating a duplicate entry in POEditor